### PR TITLE
Update ProfileSupport.java

### PR DESCRIPTION
**Pull request by berkant-k**
Default cardinality of StructureDefinition.Version is 0..1. 
So it can be empty and this code could throw NullPointerException.
We need to check the result of getVersion methot not null.

Signed-off-by: berkant-k <33718658+berkant-k@users.noreply.github.com>

### CODE REVIEW

File: fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java [hunk 1]
```diff
@@ -280,8 +280,15 @@ public static List<Constraint> getConstraints(String url, Class<?> type) {
 
     private static List<Constraint> getConstraints(StructureDefinition profile, Class<?> type) {
         String url = profile.getUrl().getValue();
-        String version = profile.getVersion().getValue();
-        CacheKey key = key(url + "|" + version);
+        CacheKey key;
```
Comment (punktilious) [inline fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java:283]: My suggestion. Use final to ensure key is assigned once and once only. Also some updates to match the style for this repo:
```
        final CacheKey key;
        if (profile.getVersion() != null) {
            key = key(url + "|" + profile.getVersion().getValue());
        } else {
            key = key(url);
        }

```

File: fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java [hunk 2]
```diff
@@ -280,8 +280,15 @@ public static List<Constraint> getConstraints(String url, Class<?> type) {
...
+        if(profile.getVersion()!=null) {
+            key = key(url + "|" + profile.getVersion().getValue());
+        }
+        else
+        {
+            key = key(url);
+        }
```
Comment (lmsurpre) [inline fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java:290]: turning robin's comment into a github suggestion:
```suggestion
        final CacheKey key;
        if(profile.getVersion()!=null) {
            key = key(url + "|" + profile.getVersion().getValue());
        } else {
            key = key(url);
        }
```

File: fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java:283]: I turned this into a github suggestion just below.  With this change, I'm happy to approve.  I'll apply it later if we don't here back.  Thanks @berkant-k !

File: fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-profile/src/main/java/org/linuxforhealth/fhir/profile/ProfileSupport.java:290]: I also added the profile.getVersion().hasValue() check which covers an extremely rare edge case where the StructureDefinition has a version, but that version has an extension but no value

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**